web/client/src/library/pages/ide/IDE.tsx

```diff
@@ -228,17 +236,23 @@ export default function PageIDE(): JSX.Element {
   }
 
   function restoreEditorTabsFromSaved(files: ModelFile[]): void {
-    files.forEach(file => {
-      if (storedTabsIds.includes(file.id)) {
-        const tab = createTab(file)
-
-        if (storedTabsId === file.id) {
-          selectTab(tab)
-        } else {
-          addTab(tab)
-        }
-      }
+    if (isArrayEmpty(storedTabs)) return
+
+    const tabs = storedTabs.map(({ id, content }) => {
+      const file = files.find(file => file.id === id) ?? createLocalFile(id)
+      const storedTab = createTab(file)
+
+      storedTab.file.content = content ?? storedTab.file.content ?? ''
+
+      return storedTab
     })
+    const tab = tabs.find(tab => tab.file.id === storedTabId)
+
+    addTabs(tabs)
+
+    if (isNil(tab)) return
+
+    selectTab(tab)
```

Comment:
Nit: I think this is slightly easier to follow but I don't feel strongly about it so feel free to ignore.
```suggestion
    if (isNotNil(tab)) {
      selectTab(tab)
    }
```